sql/codegen_xgboost.go

```diff
@@ -0,0 +1,606 @@
+	// runtime params
+	"run_local": boolPartial("run_local", func(r *xgboostFiller) *bool { return &(r.runLocal) }),
+	"workers":   uIntPartial("workers", func(r *xgboostFiller) *uint { return &(r.WorkerNum) }),
+	"memory":    uIntPartial("memory", func(r *xgboostFiller) *uint { return &(r.MemorySize) }),
+	"cpu":       uIntPartial("cpu", func(r *xgboostFiller) *uint { return &(r.CPUSize) }),
+	// booster params
+	"objective":            strPartial("objective", func(r *xgboostFiller) *string { return &(r.Objective) }),
+	"booster":              strPartial("booster", func(r *xgboostFiller) *string { return &(r.Booster) }),
+	"max_depth":            uIntPartial("max_depth", func(r *xgboostFiller) *uint { return &(r.MaxDepth) }),
+	"num_class":            uIntPartial("num_class", func(r *xgboostFiller) *uint { return &(r.NumClass) }),
+	"eta":                  fp32Partial("eta", func(r *xgboostFiller) *float32 { return &(r.Eta) }),
+	"tree_method":          strPartial("tree_method", func(r *xgboostFiller) *string { return &(r.TreeMethod) }),
+	"eval_metric":          strPartial("eval_metric", func(r *xgboostFiller) *string { return &(r.EvalMetric) }),
+	"subsample":            fp32Partial("subsample", func(r *xgboostFiller) *float32 { return &(r.Subsample) }),
+	"colsample_bytree":     fp32Partial("colsample_bytree", func(r *xgboostFiller) *float32 { return &(r.ColSampleByTree) }),
+	"colsample_bylevel":    fp32Partial("colsample_bylevel", func(r *xgboostFiller) *float32 { return &(r.ColSampleByLevel) }),
+	"max_bin":              uIntPartial("max_bin", func(r *xgboostFiller) *uint { return &(r.MaxBin) }),
+	"convergence_criteria": strPartial("convergence_criteria", func(r *xgboostFiller) *string { return &(r.ConvergenceCriteria) }),
+	"verbosity":            uIntPartial("verbosity", func(r *xgboostFiller) *uint { return &(r.Verbosity) }),
+	// xgboost train controllers
+	"num_round":  uIntPartial("num_round", func(r *xgboostFiller) *uint { return &(r.NumRound) }),
+	"auto_train": boolPartial("auto_train", func(r *xgboostFiller) *bool { return &(r.AutoTrain) }),
+	// xgboost output columns (for prediction)
+	"append_columns":  sListPartial("append_columns", func(r *xgboostFiller) *[]string { return &(r.AppendColumns) }),
+	"result_column":   strPartial("result_column", func(r *xgboostFiller) *string { return &(r.ResultColumn) }),
+	"prob_column":     strPartial("prob_column", func(r *xgboostFiller) *string { return &(r.ProbColumn) }),
+	"detail_column":   strPartial("detail_column", func(r *xgboostFiller) *string { return &(r.DetailColumn) }),
+	"encoding_column": strPartial("encoding_column", func(r *xgboostFiller) *string { return &(r.EncodingColumn) }),
+	// Label, Group, Weight and xgFeatureFields are parsed from columnClause
+}
+
+func xgParseAttr(pr *extendedSelect, r *xgboostFiller) error {
+	// parse pr.attrs to map[string][]string
+	attrs := make(map[string][]string)
+	for k, exp := range pr.attrs {
```

Comment:
Please merge the latest develop branch, I believe `pr.attrs` has been changed to `pr.trainAttrs`.